### PR TITLE
Release guidance

### PR DESCRIPTION
**Pull request by jmertic**
Brought into PR to facilitate the discussion.

### CODE REVIEW

File: process/ReleaseProcess.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Release Process and Guidelines
+
+## What is a Zowe release
+
+A release of the Zowe project consists of two 'convenience' deliverables - CLI binaries, and a PAX file which contains the Zowe API Mediation Layer, Zowe Desktop, and Zowe Explorer-Server. The CLI binaries are obtainable either through standard npm install commands, or, for users with limited internet access, through a convience zip file which allows for offline installation. The CLI convenience zip and Zowe PAX are both available through the official [Zowe Website](https://www.zowe.org).
```
Comment (jmertic) [inline process/ReleaseProcess.md:5]: Technically - these aren't Zowe open source project deliverables, but those done under the previous beta program between CA, Rocket and IBM. Right now the only official deliverables would be source code.

File: process/ReleaseProcess.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+### Zowe CLI Convenience Bundle Checklist
+
+* Confirm @next versions of the core CLI as well as plugins for packaging.
+
+### Zowe Documentation Checklist
+
+* Aggregate release notes from the Zowe subprojects, and place them in the [docs-site repository](https://github.com/zowe/docs-site).
+* Merge release notes and any other docs-site feature branches into the [docs-staging branch](https://github.com/zowe/docs-site/tree/docs-staging).
+
+### Zowe PAX Checklist
```
Comment (jmertic) [inline process/ReleaseProcess.md:34]: Just making a note this is related to the binary release question from above..

File: process/ReleaseProcess.md (hunk 1 above)
Comment (hogstrom) [reply process/ReleaseProcess.md:5]: We are continuing to create the convenience builds.  What would you recommend here?  Do you want to strike this or reword?  I would agree that the only "official" release is the source but we do deliver a large pax with installers for all the individual Zowe projects.

File: process/ReleaseProcess.md (hunk 2 above)
Comment (hogstrom) [reply process/ReleaseProcess.md:34]: I've used the term convenience build to disambiguate from source.  Is there a better term?

**Review by hogstrom — APPROVED** [review]
There are a number of good updates here and a few items that need resolution around secondary items like convenience builds.  Opening a new branch to resolve those questions

**Pull request merged**